### PR TITLE
Ravin frontend

### PR DESCRIPTION
**Pull request by RavinAnderson**
Oauth Skeleton

### CODE REVIEW

File: backend/src/routers/index.py [hunk 1]
```diff
@@ -1,11 +1,48 @@
...
 from backend.src.schemas.index import Success
 
 router = APIRouter()
 
+oauth = OAuth()
+
+CONF_URL = 'https://accounts.google.com/.well-known/openid-configuration'
+oauth.register(
+    name='google',
+    client_id="988417806604-ggnkhrhere0el8b4r3ehko3ncmt8181r.apps.googleusercontent.com",
```
Comment (ianmkim) [inline backend/src/routers/index.py:16]: @jeffreylam4 this has to be an env var. See my example in the database.py how I retrieve database credentials

File: backend/src/routers/index.py [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
 @router.get("/")
-async def index() -> Success:
-    return Success(ok=True,
-                   message = "Hello world",
-                   error = None)
+async def index(request: Request):
+    user = request.session.get('user')
+    if user:
+        name = user.get('name')
+        return HTMLResponse(f'<p>Hello {name}!</p><a href=/logout>Logout</a>')
```
Comment (ianmkim) [inline backend/src/routers/index.py:29]: we don't want this to be the index, the index should stay as is

File: backend/src/routers/index.py [hunk 3]
```diff
@@ -1,11 +1,48 @@
...
+@router.get("/login")
+async def login(request: Request):
+    redirect_uri = request.url_for('auth')
+    return await oauth.google.authorize_redirect(request, redirect_uri)
+
+@router.get("/auth")
+async def auth(request: Request):
+    token = await oauth.google.authorize_access_token(request)
+    request.session['user'] = token['userinfo']
+    # You can now create a user session or a JWT token
```
Comment (ianmkim) [inline backend/src/routers/index.py:41]: we should be checking for dartmouth edu address and create the token

File: bazel-arena [hunk 4]
```diff
@@ -1 +1 @@
-/private/var/tmp/_bazel_parvus/34f3e6d89a42d190aa96d3e97ed6faec/execroot/arena
\ No newline at end of file
+/private/var/tmp/_bazel_ravinanderson/5bdab95fb3f67d05f8432243c2b13f2e/execroot/arena
```
Comment (ianmkim) [inline bazel-arena:1]: This should be part of the .gitignore

**Review by ianmkim — CHANGES_REQUESTED** [review]

**Review by jeffreylam4 — APPROVED** [review]
Great work

**Pull request merged**